chrome/content/zotero/preferences/preferences_sync.js

```diff
@@ -177,6 +177,148 @@ Zotero_Preferences.Sync = {
+		var row = treechildren.childNodes[index];
+		var cell = row.firstChild.firstChild;
+		cell.setAttribute('value', cell.getAttribute('value') != 'true');
+
+		return this.toggleLibraryToSync(index);
+	},
+
+
+	clickLibraryToSync: function (event) {
+		var tree = document.getElementById("libraries-to-sync-tree");
+		var row = {}, col = {}, child = {};
+		tree.treeBoxObject.getCellAt(event.clientX, event.clientY, row, col, child);
+		
+		if (col.value.element.id != 'libraries-to-sync-checked') {
+			return;
+		}
+		// if clicked on checkbox update pref
+		return this.toggleLibraryToSync(row.value);
+	},
+	
+	
+	toggleLibraryToSync: function (index) {
+		var treechildren = document.getElementById('libraries-to-sync-rows');
+		if (index >= treechildren.childNodes.length) {
+			return;
+		}
+		var row = treechildren.childNodes[index];
+		var id = parseInt(row.firstChild.childNodes[1].getAttribute('value'));
+		if (isNaN(id)) {
+			return;
+		}
+		
+		var librariesToSync = JSON.parse(Zotero.Prefs.get('sync.librariesToSync') || '[]');
+		var indexOfId = librariesToSync.indexOf(id);
+		if (indexOfId != -1) {
+			librariesToSync.splice(indexOfId, 1);
+		} else {
+			librariesToSync.push(id);
+		}
+		Zotero.Prefs.set('sync.librariesToSync', JSON.stringify(librariesToSync));
+		 
+		var cell = row.firstChild.firstChild;
+		cell.setAttribute('value', indexOfId == -1);
+	},
+	
+	
+	initLibrariesToSync: Zotero.Promise.coroutine(function* () {
+		var tree = document.getElementById("libraries-to-sync-tree");
+		var treechildren = document.getElementById('libraries-to-sync-rows');
+		while (treechildren.hasChildNodes()) {
+			treechildren.removeChild(treechildren.firstChild);
+		}
+		
+		function addRow(libraryName, id, checked=false, editable=true) {
+			var treeitem = document.createElement('treeitem');
+			var treerow = document.createElement('treerow');
+			var checkboxCell = document.createElement('treecell');
+			var nameCell = document.createElement('treecell');
+			
+			nameCell.setAttribute('label', libraryName);
+			nameCell.setAttribute('value', id);
+			nameCell.setAttribute('editable', false);
+			checkboxCell.setAttribute('value', checked);
+			checkboxCell.setAttribute('editable', editable);
+			
+			treerow.appendChild(checkboxCell);
+			treerow.appendChild(nameCell);
+			treeitem.appendChild(treerow);
+			treechildren.appendChild(treeitem);
+		}
+		
+		// Add an animated row for while we're loading a group list
+		var loadingLabel = Zotero.getString("zotero.preferences.sync.librariesToSync.loadingLibraries");
+		addRow(loadingLabel, "loading", false, false);
+		var cell = treechildren.firstChild.getElementsByAttribute("value", "loading")[0];
+		function animateLoadingLabel() {
+			if (cell.getAttribute('label').length < loadingLabel.length + 3) {
+				cell.setAttribute('label', cell.getAttribute('label') + '.');
+			} else {
+				cell.setAttribute('label', loadingLabel);
+			}
+			loadTimeout = setTimeout(animateLoadingLabel, 1000.0/3.0);
+		}
```

Comment:
Clever as this is, I don't think it works — I suspect ellipses are different in different locales, and even in English three periods isn't technically typographically correct. (If we did do it, we'd also want it to start with the three instead of none, since starting with none looks like a mistake.) I think we can just add a real ellipsis to the localized term. Sorry to ruin the fun.